File: example/src/main/java/jp/wasabeef/example/recyclerview/MainActivity.java
package jp.wasabeef.example.recyclerview;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.SwitchCompat;
import android.view.View;
import android.widget.CompoundButton;

/**
 * Created by Wasabeef on 2015/03/08.
 */
public class MainActivity extends AppCompatActivity {

  private boolean enabledGrid = false;

  @Override protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);

    findViewById(R.id.btn_animator_sample).setOnClickListener(new View.OnClickListener() {
      @Override public void onClick(View v) {
        Intent i = new Intent(MainActivity.this, AnimatorSampleActivity.class);
        i.putExtra("GRID", enabledGrid);
        startActivity(i);
      }
    });

    findViewById(R.id.btn_adapter_sample).setOnClickListener(new View.OnClickListener() {
      @Override public void onClick(View v) {
        Intent i = new Intent(MainActivity.this, AdapterSampleActivity.class);
        i.putExtra("GRID", enabledGrid);
        startActivity(i);
      }
    });

    ((SwitchCompat) findViewById(R.id.grid)).setOnCheckedChangeListener(
        new CompoundButton.OnCheckedChangeListener() {
          @Override public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
            enabledGrid = isChecked;
          }
        });
  }
}

File: example/src/main/java/jp/wasabeef/example/recyclerview/MainActivity.kt
package jp.wasabeef.example.recyclerview

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.SwitchCompat

/**
 * Created by Daichi Furiya / Wasabeef on 2020/08/26.
 */
class MainActivity : AppCompatActivity() {

  companion object {
    const val KEY_GRID = "GRID"
  }

  private var enabledGrid = false

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_main)

    findViewById<View>(R.id.btn_animator_sample).setOnClickListener {
      startActivity(Intent(this, AnimatorSampleActivity::class.java).apply {
        putExtra(KEY_GRID, enabledGrid)
      })
    }

    findViewById<View>(R.id.btn_adapter_sample).setOnClickListener {
      startActivity(Intent(this, AdapterSampleActivity::class.java).apply {
        putExtra(KEY_GRID, enabledGrid)
      })
    }

    findViewById<SwitchCompat>(R.id.grid).setOnCheckedChangeListener { _, isChecked ->
      enabledGrid = isChecked
    }
  }
}
